### PR TITLE
Swarm diagnostics

### PR DESCRIPTION
**Pull request by antonyharfield**
## Summary

- `SwmUartShell` for communicating over serial with the swarm board
- `SwmConnection` for sending and parsing swarm messages (body and checksum)
- `SwmApi` for implementing specific commands - so far we need RT and DT for diagnostics
- Suite of unit tests for testing the connection, checksums and API
- Local database for storing swarm diagnostics
- Swarm services adapted from sbd services (somewhat untested?)

### CODE REVIEW

File: role-admin/src/main/java/org/rfcx/guardian/admin/RfcxGuardian.java [hunk 1]
```diff
@@ -151,12 +154,13 @@
...
-				SmsDispatchCycleService.SERVICE_NAME,
-				SbdDispatchCycleService.SERVICE_NAME,
-				SwmDispatchCycleService.SERVICE_NAME,
-				CompanionSocketService.SERVICE_NAME
+					DeviceSystemService.SERVICE_NAME,
+					DeviceI2cService.SERVICE_NAME,
+					SmsDispatchCycleService.SERVICE_NAME,
+					SbdDispatchCycleService.SERVICE_NAME,
+					SwmDispatchCycleService.SERVICE_NAME,
+					SwmDiagnosticService.SERVICE_NAME,
```
Comment (antonyharfield) [inline role-admin/src/main/java/org/rfcx/guardian/admin/RfcxGuardian.java]: @Tooseriuz Are we sure this is a core service in the architecture that we are following? I suspect there should only be one `Swm` service here.

File: role-guardian/src/main/java/org/rfcx/guardian/guardian/asset/meta/MetaJsonUtils.java [hunk 2]
```diff
@@ -72,6 +72,11 @@ public void createSystemMetaDataJsonSnapshot() throws JSONException {
 				"sentinel_sensor", app.getResolver()));
 		if (sentinelSensor.length() > 0) { metaDataJsonObj.put("sentinel_sensor", sentinelSensor); }
 
+		// Adding Swarm diagnostic data, if they can be retrieved from admin role via content provider
+		String swmDiagnostic = getConcatMetaField(RfcxComm.getQuery("admin", "database_get_all_rows",
+				"swm_meta", app.getResolver()));
+		if (swmDiagnostic.length() > 0) { metaDataJsonObj.put("swm_meta", swmDiagnostic); }
```
Comment (antonyharfield) [inline role-guardian/src/main/java/org/rfcx/guardian/guardian/asset/meta/MetaJsonUtils.java]: Should we call it `swm_diagnostics` or just `swm`? (It's already inside meta blob, so adding _meta seems redundant.)

Comment (Tooseriuz) [reply role-guardian/src/main/java/org/rfcx/guardian/guardian/asset/meta/MetaJsonUtils.java]: Okay - that makes more sense.

File: .idea/gradle.xml [hunk 3]
```diff
@@ -4,10 +4,10 @@
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
-        <option name="delegatedBuild" value="false" />
         <option name="testRunner" value="GRADLE" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
+        <option name="gradleJvm" value="11" />
```
Comment (antonyharfield) [inline .idea/gradle.xml:10]: This doesn't work for me. Can we remove it?

**Review by Tooseriuz — APPROVED** [review]

**Pull request merged**